homeassistant/components/husqvarna_automower/coordinator.py

```diff
@@ -69,19 +69,18 @@ async def client_listen(
         """Listen with the client."""
         try:
             await automower_client.auth.websocket_connect()
-            reconnect_time = 2
-            await automower_client.start_listening()
         except HusqvarnaWSServerHandshakeError as err:
             _LOGGER.debug(
                 "Failed to connect to websocket. Trying to reconnect: %s", err
             )
-
-        if not hass.is_stopping:
-            await asyncio.sleep(reconnect_time)
-            reconnect_time = min(reconnect_time * 2, MAX_WS_RECONNECT_TIME)
-            await self.client_listen(
-                hass=hass,
-                entry=entry,
-                automower_client=automower_client,
-                reconnect_time=reconnect_time,
-            )
+            if not hass.is_stopping:
+                reconnect_time = 2
+                await asyncio.sleep(reconnect_time)
```

Comment:
We may need to store the `reconnect_time` in an instance variable so we can reset it after a successful connection. Although, we will never retry now if the connection is successful even if the listening call fails. We should probably retry again in that case too. Otherwise the websocket connection will never recover until the config entry is reloaded.